src/x/sync/pooled_worker_pool.go

```diff
@@ -99,8 +108,28 @@ func (p *pooledWorkerPool) Go(work Work) {
 	}
 
 	if !p.growOnDemand {
-		workCh <- work
-		return
+		if timeout <= 0 {
+			workCh <- work
+			return true
+		}
+
+		// Attempt to try writing without allocating a ticker.
+		select {
+		case workCh <- work:
```

Comment:
so this timeout doesn't ensure the work completes by the deadline, just the work starts?

Reply:
Yes true hah, it will not perform any cancellation.